sklearn/ensemble/tests/test_forest.py

```diff
@@ -947,6 +947,13 @@ def check_class_weights(name):
     clf2.fit(iris.data, iris.target, sample_weight)
     assert_almost_equal(clf1.feature_importances_, clf2.feature_importances_)
 
+    # When sample_weights is an unsupported array type, it checks if it raises
+    # an exception. With Python 2.x lists it complains there is no copy()
+    # method. On success the list should be automatically converted.
+    clf = ForestClassifier()
+    sample_weight = [1.] * len(iris.data)
+    clf.fit(iris.data, iris.target, sample_weight=sample_weight)
```

Comment:
You need to add a test that fails on master to show that your PR is actually the right fix. AFAICT this one does not fail on master.